include/micm/solver/backward_euler.inl

```diff
@@ -0,0 +1,205 @@
+    // y_{n+1} = y_n + H * f(t_{n+1}, y_{n+1})
+    // This is a root finding problem because you need to know y_{n+1} to compute f(t_{n+1}, y_{n+1})
+    // you need to solve the equation y_{n+1} - y_n - H f(t_{n+1}, y_{n+1}) = 0
+    // We will also use the same logic used by cam-chem to determine the time step
+    // That scheme is this: 
+    // Start with H = time_step
+    // if that fails, try H = H/2
+    // if that fails, try H = H/10
+    // if that fails, return whatever the current integration with the current H is
+
+    double H = time_step;
+    double t = 0.0;
+    double tol = 1e-4;
+    double small = 1.0e-40;
+    auto temp = state.variables_;
+    auto forcing = state.variables_;
+    auto residual = state.variables_;
+    bool singular = false;
+    std::size_t max_iter = 11;
+    std::size_t n_successful_integrations = 0;
+    std::size_t n_convergence_failures = 0;
+    std::size_t cut_in_half_limit = 4;
+    std::size_t cut_in_tenth_limit = 1;
+    const std::array<double, 5> time_step_reductions{ 0.5, 0.5, 0.5, 0.5, 0.1 };
+
+    auto Yn = state.variables_;
+    auto Yn1 = state.variables_;
+
+    Process::UpdateState(processes, state);
+
+    while(t < time_step) {
+      bool converged = true;
+      std::size_t iterations = 0;
+
+      do {
+        // calculate forcing
+        std::fill(forcing.AsVector().begin(), forcing.AsVector().end(), 0.0);
+        process_set.AddForcingTerms(state.rate_constants_, Yn, forcing);
+
+        // calculate jacobian
+        std::fill(state.jacobian_.AsVector().begin(), state.jacobian_.AsVector().end(), 0.0);
+        process_set.SubtractJacobianTerms(state.rate_constants_, Yn, state.jacobian_);
+
+        // subtract the inverse of the time step from the diagonal
+        // TODO: handle vectorized jacobian matrix
+        for(auto& jac : state.jacobian_.AsVector()) {
+          jac *= -1;
+        }
+        for (std::size_t i_block = 0; i_block < state.jacobian_.Size(); ++i_block)
+        {
+          auto jacobian_vector = std::next(state.jacobian_.AsVector().begin(), i_block * state.jacobian_.FlatBlockSize());
+          for (const auto& i_elem : jacobian_diagonal_elements)
+            jacobian_vector[i_elem] -= 1 / H;
+        }
+
+        // We want to solve this equation for a zero
+        // (y_{n+1} - y_n) / H = f(t_{n+1}, y_{n+1})
+
+        // try to find the root by factoring and solving the linear system
+        linear_solver.Factor(state.jacobian_, state.lower_matrix_, state.upper_matrix_, singular);
+
+        auto yn1_iter = Yn1.begin();
+        auto yn_iter = Yn.begin();
+        auto residual_iter = residual.begin();
+        auto forcing_iter = forcing.begin();
+        // forcing_blk in camchem
+        // residual = (Yn1 - Yn) / H - forcing;
+        for(; yn1_iter != Yn1.end(); ++yn1_iter, ++yn_iter, ++residual_iter, ++forcing_iter) {
+          *residual_iter = (*yn1_iter - *yn_iter) / H - *forcing_iter;
+        }
```

Comment:
This should be replaced by:
```
        for(; yn_iter != Yn.end(); ++yn_iter, ++iter_invariant_iter, ++residual_iter, ++forcing_iter) {
          *residual_iter = *yn_iter / H - *iter_invariant_iter - *forcing_iter;
        }
```
Both `yn1` and `yn` are updated during the Newton iterations, but we want to subtract a constant term instead here.